### PR TITLE
fix(opStack): Maybe add proof node

### PR DESCRIPTION
**Pull request by roninjin10**
Fixes vulnerability where an attacker could make all proofs start failing.  See https://github.com/ethereum-optimism/optimism/pull/9663

---

## PR-Codex overview
This PR focuses on adding proof validation for `buildProveWithdrawal` in OP Stack. 

### Detailed summary
- Added proof validation for `buildProveWithdrawal`
- Introduced `maybeAddProofNode` function
- Modified `buildProveWithdrawal` to handle proof nodes
- Added tests for proof node handling and proof appending

> The following files were skipped due to too many changes: `src/chains/opStack/actions/buildProveWithdrawal.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/chains/opStack/actions/buildProveWithdrawal.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import type { Address } from 'abitype'
+import { type Hex, fromRlp, toRlp } from '~viem/index.js'
```
Comment (roninjin10) [inline src/chains/opStack/actions/buildProveWithdrawal.ts:2]: fixme

**Comment by roninjin10** [comment]
@smartcontracts

File: src/chains/opStack/actions/buildProveWithdrawal.ts [hunk 2]
```diff
@@ -138,7 +181,7 @@ export async function buildProveWithdrawal<
       version: outputRootProofVersion,
     },
     targetChain: chain,
-    withdrawalProof: proof.storageProof[0].proof,
+    withdrawalProof: maybeAddProofNode(slot, proof.storageProof[0].proof),
```
Comment (smartcontracts) [inline src/chains/opStack/actions/buildProveWithdrawal.ts:184]: Need to keccak256(slot) here

**Review by smartcontracts — APPROVED** [review]
Noice

**Pull request merged**